vault-core/src/main/java/org/apache/jackrabbit/vault/fs/api/WorkspaceFilter.java

```diff
@@ -102,6 +103,19 @@ public interface WorkspaceFilter extends Dumpable {
      */
     boolean isAncestor(@NotNull String path);
 
+    /**
+     * Matches the given path with all filter roots. For each, if it is an ancestor,
+     * add the name of the first path segment of the remaining filter root "below" path
+     * to the result set.
+     *
+     * @param path Path to check
+     * @return first path segments of non-matched paths, or {@code null} when result set
```

Comment:
The condition when null is returned is not really clear. What if only some roots are no offsprings of the given path?

Reply:
It would return null, because in that case we can't optimize? So clarify the docs?

Reply:
Attempted to clariffy, please check.

(Note that there may be room for improvements here, but  for now the change is sufficient for my use case)

Reply:
For me there is three conditions:

1. all filter roots are below given path
2. a filter root is not related (i.e. neither ancestor nor descendent)
3. a filter root is an ancestor of this path

Are you sure that both 2. and 3. should lead to the same return value `null`?

Reply:
Well - `null` is the safe approach, because it means doing exactly what we did before.

And no, I'm not sure. I'm trying to improve one specific use case (for which I happen to have an integration test to verify). We probably can improve more cases, but my personal preference would be to do that when we figure out that it's needed - for instance when the current improvement can't be applied because there is a second completely unrelated filter.

Does this make sense?